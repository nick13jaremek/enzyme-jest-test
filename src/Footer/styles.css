.footer {
  max-height: 200px;
  background-color: orange;
  width: 100%;
};

