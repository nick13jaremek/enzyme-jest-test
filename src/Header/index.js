import React from 'react';

const Header = () => {
  return (
    <div className="header">
      <h1>Jest + Enzyme!</h1>
    </div>
  );
};

export default Header;

