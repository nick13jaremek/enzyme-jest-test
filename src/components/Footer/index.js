import React from 'react';

const Footer = () => {
  return (
    <div className="footer">
      <ul>
        <li>Contact</li>
        <li>About</li>
      </ul>
    </div>
  );
};

export default Footer;
