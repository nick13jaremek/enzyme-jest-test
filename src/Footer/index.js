import React from 'react';
import './styles.css';

const Footer = () => {
  return (
    <div className="footer">
      <ul>
        <li>Contact</li>
        <li>About</li>
      </ul>
    </div>
  );
};

export default Footer;
