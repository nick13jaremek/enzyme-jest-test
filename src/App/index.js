import React, { Component } from 'react';
import './styles.css';
import Header from '../Header';
import Footer from '../Footer';

class App extends Component {
  render() {
    return (
      <div className="App">
      <Header />
      <Footer />
      </div>
    );
  }
}

export default App;
