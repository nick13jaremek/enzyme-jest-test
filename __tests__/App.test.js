import React from 'react';
import { shallow, mount } from 'enzyme';
import App from '../src/App';
import Header from '../src/Header';
import Footer from '../src/Footer';

describe('App component', () => {
  let wrapper;

  describe('shallow', () => {
    beforeEach(() => {
      wrapper = shallow(<App />);
    });

    test('should match snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });

    test('should have mounted the component', () => {
     expect(wrapper).toBeDefined(); 
    });

    test('should have a Header component', () => {
      expect(wrapper.find(Header)).toHaveLength(1);
    });

    test('should have a Footer component', () => {
      expect(wrapper.find(Footer)).toHaveLength(1);
    });


    test('should not find the header component class name', () => {
      expect(wrapper.find('.header')).toHaveLength(0);
    });

    test('should not find the footer component class name', () => {
      expect(wrapper.find('.footer')).toHaveLength(0);
    });
  });

  describe('mount', () => {
    beforeEach(() => {
      wrapper = mount(<App />);
    });

    test('should match snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });

    test('should have mounted the component', () => {
      expect(wrapper).toBeDefined();
    });

    test('should have a Header component wrapping the actual component markup code', () => {
      console.log(wrapper.debug());
      expect(wrapper.find(Header)).toHaveLength(1);

      expect(wrapper.find('.header')).toHaveLength(1);
    });

    test('should have a Footer component as-is', () => {
      expect(wrapper.find(Footer)).toHaveLength(1);

      expect(wrapper.find('.footer')).toHaveLength(1);
    });
  });
});
