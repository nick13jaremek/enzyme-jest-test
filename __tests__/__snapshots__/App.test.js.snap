// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component mount should match snapshot 1`] = `
<App>
  <div
    className="App"
  >
    <Header>
      <div
        className="header"
      >
        <h1>
          Jest + Enzyme!
        </h1>
      </div>
    </Header>
    <Footer>
      <div
        className="footer"
      >
        <ul>
          <li>
            Contact
          </li>
          <li>
            About
          </li>
        </ul>
      </div>
    </Footer>
  </div>
</App>
`;

exports[`App component shallow should match snapshot 1`] = `
<div
  className="App"
>
  <Header />
  <Footer />
</div>
`;
